gobblin-runtime/src/main/java/gobblin/runtime/instance/DefaultGobblinInstanceDriverImpl.java

```diff
@@ -188,7 +189,7 @@ public void run() {
          JobExecutionDriver driver = _jobLauncher.launchJob(_jobSpec);
          _callbacksDispatcher.onJobLaunch(driver);
          driver.registerStateListener(new JobStateTracker());
-         driver.startAsync();
+        new Thread(driver).start();
```

Comment:
Can you use gobblin.util.ExecutorsUtils.newThreadFactory(Optional<Logger>, Optional<String>) so that we can name the thread for debugging purposes and we log uncaught exceptions?